packages/customer/src/models/address.ts

```diff
@@ -0,0 +1,88 @@
+import { DAL } from "@medusajs/types"
+import { generateEntityId } from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Entity,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+  ManyToOne,
+} from "@mikro-orm/core"
+import Customer from "./customer"
+
+type OptionalAddressProps = DAL.EntityDateColumns // TODO: To be revisited when more clear
+
+@Entity({ tableName: "customer_address" })
+export default class Address {
+  [OptionalProps]: OptionalAddressProps
+
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @Property({ columnType: "text" })
+  customer_id: string
+
+  @ManyToOne(() => Customer, {
+    fieldName: "customer_id",
+  })
+  customer?: Customer
+
+  @Property({ columnType: "text", nullable: true })
+  company?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  first_name?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  last_name?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  address_1?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  address_2?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  city?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  country_code?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  province?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  postal_code?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  phone?: string | null
+
+  @Property({ columnType: "jsonb", nullable: true })
+  metadata?: Record<string, unknown> | null
```

Comment:
```suggestion
  @Property({ columnType: "text", nullable: true })
  company: string | null = null

  @Property({ columnType: "text", nullable: true })
  first_name: string | null = null

  @Property({ columnType: "text", nullable: true })
  last_name: string | null = null

  @Property({ columnType: "text", nullable: true })
  address_1: string | null = null

  @Property({ columnType: "text", nullable: true })
  address_2: string | null = null

  @Property({ columnType: "text", nullable: true })
  city: string | null = null

  @Property({ columnType: "text", nullable: true })
  country_code: string | null = null

  @Property({ columnType: "text", nullable: true })
  province: string | null = null

  @Property({ columnType: "text", nullable: true })
  postal_code: string | null = null

  @Property({ columnType: "text", nullable: true })
  phone: string | null = null

  @Property({ columnType: "jsonb", nullable: true })
  metadata: Record<string, unknown> | null = null
```

Reply:
question: again I have based this off the CartModule - is there a thread somewhere that explains this pattern so I can just understand the details it? :)

Reply:
ahah sorry 😄 yes sure, we had a discussion in the end channel and we added it in our module convention [here](https://www.notion.so/medusajs/Modules-Code-Conventions-7ed058a9fc56448cafd3f0ccd1a9ad73)

Reply:
cc @olivermrbl will you update the cart in follow up pr's?